magda-gateway/src/reloadTenants.ts

```diff
@@ -0,0 +1,31 @@
+import { Tenant } from "@magda/typescript-common/dist/generated/registry/api";
+import * as request from "request";
+import { registryApi } from "./setupTenantMode";
+import { MAGDA_ADMIN_PORTAL_ID } from "@magda/typescript-common/dist/registry/TenantConsts";
+
+export const tenantsTable = new Map<String, Tenant>();
+
+export default async function reloadTenants() {
```

Comment:
Can you explain that in a bit more detail?

Do you mean that there's only likely to be incoming requests with `Host` set to values in already in the DB? That's true, but there's still potential for someone to point a domain to the Magda instance and forget to set up the corresponding tenant record - if there's a large amount of traffic coming through that domain, it might overload the database, which would in turn prevent anyone from setting up the tenant record to fix it 😬 .

Reply:
I thought the magda admin was responsible for pointing a domain to the Magda instance and should never forget to add the corresponding tenant record. However, the potential error could happen, I agree that adding throttle is a good idea. Will do it.

Reply:
done

Reply:
Re-implemented and added full tests.